#YS486-1
./call-variants.sh -a '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/YS486-1_S49_L001_R1_001.fastq.gz' -b '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/YS486-1_S49_L002_R1_001.fastq.gz' -c '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/YS486-1_S49_L001_R2_001.fastq.gz' -d '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/YS486-1_S49_L002_R2_001.fastq.gz' -s 'YS486-1' -p '/projects/home/agombolay3/data/bin/Trimmomatic-0.36'

#YS486-2
./call-variants.sh -a '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/YS486-2_S58_L001_R1_001.fastq.gz' -b '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/YS486-2_S58_L002_R1_001.fastq.gz' -c '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/YS486-2_S58_L001_R2_001.fastq.gz' -d '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/YS486-2_S58_L002_R2_001.fastq.gz' -s 'YS486-2' -p '/projects/home/agombolay3/data/bin/Trimmomatic-0.36'

#CM3
./call-variants.sh -a '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM3_S50_L001_R1_001.fastq.gz' -b '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM3_S50_L002_R1_001.fastq.gz' -c '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM3_S50_L001_R2_001.fastq.gz' -d '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM3_S50_L002_R2_001.fastq.gz' -s 'CM3' -p '/projects/home/agombolay3/data/bin/Trimmomatic-0.36'

#CM6
./call-variants.sh -a '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM6_S51_L001_R1_001.fastq.gz' -b '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM6_S51_L002_R1_001.fastq.gz' -c '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM6_S51_L001_R2_001.fastq.gz' -d '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM6_S51_L002_R2_001.fastq.gz' -s 'CM6' -p '/projects/home/agombolay3/data/bin/Trimmomatic-0.36'

#CM9
./call-variants.sh -a '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM9_S52_L001_R1_001.fastq.gz' -b '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM9_S52_L002_R1_001.fastq.gz' -c '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM9_S52_L001_R2_001.fastq.gz' -d '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM9_S52_L002_R2_001.fastq.gz' -s 'CM9' -p '/projects/home/agombolay3/data/bin/Trimmomatic-0.36'

#CM10
./call-variants.sh -a '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM10_S53_L001_R1_001.fastq.gz' -b '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM10_S53_L002_R1_001.fastq.gz' -c '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM10_S53_L001_R2_001.fastq.gz' -d '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM10_S53_L002_R2_001.fastq.gz' -s 'CM10' -p '/projects/home/agombolay3/data/bin/Trimmomatic-0.36'

#CM11
./call-variants.sh -a '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM11_S54_L001_R1_001.fastq.gz' -b '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM11_S54_L002_R1_001.fastq.gz' -c '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM11_S54_L001_R2_001.fastq.gz' -d '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM11_S54_L002_R2_001.fastq.gz' -s 'CM11' -p '/projects/home/agombolay3/data/bin/Trimmomatic-0.36'

#CM12
./call-variants.sh -a '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM12_S55_L001_R1_001.fastq.gz' -b '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM12_S55_L002_R1_001.fastq.gz' -c '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM12_S55_L001_R2_001.fastq.gz' -d '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM12_S55_L002_R2_001.fastq.gz' -s 'CM12' -p '/projects/home/agombolay3/data/bin/Trimmomatic-0.36'

#CM41
./call-variants.sh -a '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM41_S56_L001_R1_001.fastq.gz' -b '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM41_S56_L002_R1_001.fastq.gz' -c '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM41_S56_L001_R2_001.fastq.gz' -d '/projects/home/agombolay3/data/repository/Variant-Calling-Project/Sequencing-Results/CM41_S56_L002_R2_001.fastq.gz' -s 'CM41' -p '/projects/home/agombolay3/data/bin/Trimmomatic-0.36'
