# Helpful Resources

* Variant Calling
     * [HaplotypeCaller](https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeCaller.php)
     * [GenotypeGVCFs](https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeGVCFs.php)
     * [Known variants/sites](https://software.broadinstitute.org/gatk/documentation/article.php?id=1247)

* [VCF file Format](http://www.internationalgenome.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-40/)

* Filter variants in VCF
     * [SNPeff](http://snpeff.sourceforge.net/)
     * [SNPSift](http://snpeff.sourceforge.net/SnpSift.html)
     * [Tutorial](http://ddocent.com/filtering/)
     
* [Calling variants on cohorts of samples using the HaplotypeCaller in GVCF mode](https://software.broadinstitute.org/gatk/documentation/article.php?id=3893)

* [What does one do with the raw VCF output of joint genotyping many gVCF files?](http://gatkforums.broadinstitute.org/gatk/discussion/7478/what-does-one-do-with-the-raw-vcf-output-of-joint-genotyping-many-gvcf-files)
