# Helpful Resources

* Variant Calling  
      * [HaplotypeCaller](https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeCaller.php)
      * [GenotypeGVCFs](https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeGVCFs.php)
      * [Known variants/sites](https://software.broadinstitute.org/gatk/documentation/article.php?id=1247)


* [De novo assembly](https://github.com/lexnederbragt/INF-BIO9120_fall2013_de_novo_assembly/blob/master/practicals/02_Mapping_reads_to_an_assembly.md)
      * [SPADES](http://cab.spbu.ru/software/spades/)
      * [Assembly using SPADES](http://inf-biox121.readthedocs.io/en/2015/Assembly/practicals/06_Assembly_using_SPADES.html)

* Filter variants in VCF
      * [SNPSift]()
         * [Filter](http://snpeff.sourceforge.net/SnpSift.html#filter)
         * [Extract](http://snpeff.sourceforge.net/SnpSift.html#Extract)
      * [SNPeff](http://snpeff.sourceforge.net/SnpSift.html)
      * [VCF Format](http://www.internationalgenome.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-40/)
      * [SNP Filtering Tutorial](http://ddocent.com/filtering/)
    
